packages/arb-token-bridge-ui/src/pages/api/arbstats.ts

```diff
@@ -0,0 +1,95 @@
+// Connects to Arbiscan and scrapes some nice info (currently only TPS) which we can show on our UI as well
+
+import axios from 'axios'
+import { load } from 'cheerio'
+import { NextApiRequest, NextApiResponse } from 'next'
+
+import { getExplorerUrl, isNetwork } from 'src/util/networks'
+
+// Extending the standard NextJs request with Deposit-params
+type NextApiRequestWithArbStatsParams = NextApiRequest & {
+  query: {
+    l2ChainId: string
+  }
+}
+
+export type ArbStats = {
+  tps: number | null
+}
+
+type StatsResponse = {
+  data: ArbStats
+  message?: string // in case of any error
+}
+
+const emptyStats: ArbStats = { tps: null }
+
+export default async function handler(
+  req: NextApiRequestWithArbStatsParams,
+  res: NextApiResponse<StatsResponse>
+) {
+  try {
+    // validate method
+    if (req.method !== 'GET') {
+      res
+        .status(400)
+        .send({ message: `invalid_method: ${req.method}`, data: emptyStats })
+      return
+    }
+
+    const { l2ChainId } = req.query
+
+    // validate the request parameters
+    const errorMessage = []
+    if (!l2ChainId) errorMessage.push('<l2ChainId> is required')
+
+    if (errorMessage.length) {
+      res.status(400).json({
+        message: `incomplete request: ${errorMessage.join(', ')}`,
+        data: emptyStats
+      })
+    }
+
+    // url from where we'll fetch stats
+    const explorerUrl = getExplorerUrl(Number(l2ChainId))
+
+    // if the network is not ArbOne or Nova, or no explorer URL then we don't have the required stats on the explorer
+    // don't unnecessarily call fetch function. return empty stats.
+    if (
+      isNetwork(Number(l2ChainId)).isTestnet ||
+      !isNetwork(Number(l2ChainId)).isSupported ||
+      !explorerUrl
```

Comment:
Can't we just check if the network is either Arbitrum One or Nova?

Reply:
Updated. 
There are 3 parts to it now. 
- We will send a `400 bad request` if the network chain id is not passed.
- We will send a `400 bad request` if the network passed is not supported by our bridge at all. 
- We will send `200 OK`, but with null data in case the network passed is not supported by our TPS call. 

I did this because going forward, our stats might have a lot more info than TPS, so it is not ideal to return 400 just because we cannot get the network's TPS right now.
Also, if the network type is supported by the bridge, but not by stats call, we were getting `400 error` in stats call every time the user switched to even Goerli which was not ideal.